datafusion/sql/src/expr/mod.rs

```diff
@@ -704,7 +706,20 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
         };
         Ok(Expr::ScalarFunction(ScalarFunction::new(fun, args)))
     }
-
+    fn sql_position_to_expr(
+        &self,
+        substr_expr: SQLExpr,
+        str_expr: SQLExpr,
+        schema: &DFSchema,
+        planner_context: &mut PlannerContext,
+    ) -> Result<Expr> {
+        let fun = BuiltinScalarFunction::Position;
```

Comment:
I wonder would it be possible here to reuse the existing `BuiltinScalarFunction::Instr` rather than creating a new function for `position`?

Reply:
I think there is no difference between the two functions, but I just browsed the previous implementation, each scalarbilutin function would have a corresponding implementation in string_expression, I don't know whether it is good to maintain the consistency or not. I can definitely reuse instr though.

Reply:
I think we should reuse `Instr` unless there is a compelling reason to have a new function

Reply:
OK, I will fix it soon

Reply:
THank you 🙏

Reply:
I think if we changed this code to use `BuiltinScalarFunction::Instr` we could remove `BuiltinScalarFunction::Position` entirely which would make this PR much simpler and address @Weijun-H and my concerns about duplication

What do you think @Lordworms

Reply:
> YES, I would do this right now, sorry I have been working on another issue

No rush -- sorry I want to make it clear there is no pressure here. I just wanted to make sure your PR wasn't waiting on feedback (as I get anxious when there are so many PRs waiting on review)

Thank you for all your help